RoddyCore/test/src/de/dkfz/roddy/execution/jobs/EffectiveToolCommandBuilderSpec.groovy

```diff
@@ -0,0 +1,237 @@
+        ctx.outputGroupString >> "someGroup"
+        ctx.wrapInCommand >> someWrapper
+        EffectiveToolCommandBuilder builder = EffectiveToolCommandBuilder.from(ctx)
+
+        when:
+        Optional<ToolCommand> result = builder.build(someToolCommand)
+
+        then:
+        result.isPresent()
+        result.get().command instanceof Command
+        (result.get().command as Command).toCommandSegmentList() == [u("someWrapper")]
+    }
+
+    def "sg with apptainer"() {
+        given:
+        ExecutionContext ctx = Mock(ExecutionContext)
+        ctx.jobExecutionEnvironmentOrBash >> JobExecutionEnvironment.apptainer
+        ctx.outputGroupString >> "someGroup"
+        ctx.userContainerEngineArguments >> []
+        ctx.containerEnginePath >> Paths.get("apptainer")
+        ctx.roddyContainerCopyVariables >> []
+        ctx.containerImage >> "someImage"
+        ctx.roddyMounts >> []
+        ctx.userContainerMounts >> [
+                new BindSpec(Paths.get("/hostPath"),
+                             Paths.get("/containerPath"),
+                             BindSpec.Mode.RW)
+        ]
+        ctx.outputDirectory >> new File("/tmp")
+        ctx.wrapInCommand >> someWrapper
+        EffectiveToolCommandBuilder builder = EffectiveToolCommandBuilder.from(ctx)
+
+        when:
+        Optional<ToolCommand> result = builder.build(someToolCommand)
+
+        then:
+        result.isPresent()
+        result.get().command instanceof Code
+        // It is important for DefaultPlugin < 1.3 that `sgWasCalled=true` and for < 1.2.2-4 that
+        // `newGrpIsCalled=true`. Newer versions of the plugin compare the actual primary group
+        // against the desired target group.
+        forBash((result.get().command as Code).toEscapableString(true)) ==
+                "#!/bin/bash\n" +
+                "sg someGroup -c apptainer\\ exec\\ -B\\ /hostPath:/containerPath:rw\\ --env\\ sgWasCalled\\=true\\ --env\\ newGrpIsCalled\\=true\\ -W\\ /tmp\\ someImage\\ someWrapper\n"
+    }
+
+    def "no sg with apptainer()"(toolCommand) {
+        given:
+        ExecutionContext ctx = Mock(ExecutionContext)
+        ctx.jobExecutionEnvironmentOrBash >> JobExecutionEnvironment.apptainer
+        ctx.userContainerEngineArguments >> []
+        ctx.containerEnginePath >> Paths.get("apptainer")
+        ctx.roddyContainerCopyVariables >> []
+        ctx.containerImage >> "someImage"
+        ctx.roddyMounts >> []
+        ctx.userContainerMounts >> []
+        ctx.outputDirectory >> new File("/tmp")
+        ctx.wrapInCommand >> someWrapper
+        EffectiveToolCommandBuilder builder = EffectiveToolCommandBuilder.from(ctx)
+
+        when:
+        Optional<ToolCommand> result = builder.build(toolCommand)
+
+        then:
+        result.isPresent()
+        result.get().command instanceof Code
+        // It is OK to set `sgWasCalled=false` because the primary group is not being changed.
```

Comment:
as an FYI: you can also do this as:
`and: 'It is OK to set \'sgWasCalled=false\' because the primary group is not being changed.'`
not sure if it is a good idea here, but just in general